### PR TITLE
Integrate prettier and fix tslint script

### PR DESCRIPTION
**Pull request by vishal423**
- Integrate `prettier`
- Fix `tslint` script configurations
- Updated `tslint` configurations and disabled conflicting rules.
- Auto format code with prettier, executed lint and lint fix

### CODE REVIEW

**Comment by vishal423** [comment]
@deepu105, @wmarques , WDYT?

**Comment by vishal423** [comment]
Can someone please integrate `sonar` on this project?

File: .prettierrc.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "useTabs": false,
+  "printWidth": 140,
+  "tabWidth": 2,
+  "singleQuote": true,
+  "trailingComma": "none",
+  "arrowParens": "avoid"
+}
```
Comment (wmarques) [inline .prettierrc.json]: The useTabs, arrowParens and trailingComma are at default values so I don't think it's worth to add theme here.

File: package.json [hunk 2]
```diff
@@ -27,7 +27,8 @@
     "@ngx-translate/http-loader": "^4.0.0",
     "@fortawesome/angular-fontawesome": "^0.3.0",
     "@fortawesome/fontawesome-svg-core": "^1.2.12",
-    "@fortawesome/free-solid-svg-icons": "^5.6.3"
+    "@fortawesome/free-solid-svg-icons": "^5.6.3",
+    "rxjs": "^6.3.3"
```
Comment (wmarques) [inline package.json]: why is rxjs added there?

File: package.json [hunk 3]
```diff
@@ -79,9 +83,16 @@
...
+  "lint-staged": {
+    "{,src/**/}*.{md,json,ts}": [
+      "prettier --write",
+      "git add"
+    ]
+  },
   "scripts": {
-    "lint": "tslint --project tslint.json",
-    "lint:fix": "tslint --project tslint.json --fix",
+    "format": "prettier --write \"{,src/**/}*.{md,json,ts}\"",
```
Comment (wmarques) [inline package.json]: add also html file please

File: package.json [hunk 4]
```diff
@@ -79,9 +83,16 @@
       "<rootDir>/tests/**/*.spec.ts"
     ]
   },
+  "lint-staged": {
+    "{,src/**/}*.{md,json,ts}": [
```
Comment (wmarques) [inline package.json]: add also html extension.
Remove the `{}` around `{,src/**/`

File: .prettierrc.json (hunk 1 above)
Comment (vishal423) [reply .prettierrc.json]: sure.. i just took default values from main `generator-jhipster`. I will check again and update.

File: package.json (hunk 3 above)
Comment (vishal423) [reply package.json]: I think we don't have any `.html` files in this library.

File: package.json (hunk 2 above)
Comment (vishal423) [reply package.json]: Rxjs is being used @ 
https://github.com/jhipster/ng-jhipster/blob/9e10c4fae4bb3c5643dfe6f40ef91b460357b41f/src/service/event-manager.service.ts#L20

File: package.json (hunk 4 above)
Comment (vishal423) [reply package.json]: `{,src/**/}*` is to match files in the current directory (see comma before `src`) and in `src` directory.

File: package.json (hunk 3 above)
Comment (wmarques) [reply package.json]: Yes but in the future we can have some

Comment (vishal423) [reply package.json]: updated

File: package.json (hunk 2 above)
Comment (wmarques) [reply package.json]: Then we should add `rxjs` as a `devDependencies` also :)

Comment (wmarques) [reply package.json]: Oh sorry didn't see !
Can you then please add the same `~` selector ? To be consistent between the dev and peer dependencies :)

Comment (vishal423) [reply package.json]: updated to use caret in dev dependencies

**Comment by wmarques** [comment]
LGTM.
@vishal423 Did u test this on some generated apps ?

**Comment by vishal423** [comment]
@wmarques, I did basic sanity and things seems to work. During testing, I noted an issue related to `JhiKeys` in the metrics component, but, that's out of scope of this work and should be addressed separately.

**Comment by wmarques** [comment]
LGTM then :) 
@deepu105 Do you want to check this or i merge ?

**Comment by deepu105** [comment]
@wmarques I trust you so its fine by me

**Pull request merged**